### PR TITLE
Add gitops-update Pipeline parameter to support a custom target git branch

### PR DESCRIPTION
**Pull request by LaVLaS**
## Description
Introduce a parameter to the `gitop-update-pipeline` to allow users to specify a target branch name for the automated PR pipeline.  The default value is still `main` but this will allow a `PipelineRun` to target custom branch like `my-edge-demo`

## How Has This Been Tested?
I deployed this fix locally as part of the ai-edge demo that was using a new branch that was created from `main`

## Merge criteria:
No issues with the parameter and the documentation is clear

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: pipelines/README.md [hunk 1]
```diff
@@ -175,6 +175,14 @@ The `tekton/gitops-update-pipeline/example-pipelineruns/` contains some examples
 In these examples, notice that there is a template Secret file for the Git credentials that are referenced by different tasks.
 Create an equivalent Secret with appropriate details for your environment, and change the parameter values in the PipelineRun definition to match.
 
+NOTE: If you are using a branch other than `main` for your gitops workflow, you will need to modify the `PipelineRuns` examples under `tekton/gitops-update-pipeline/example-pipelineruns/` to set the `gitRepoBranchBase` parameter to your custom branch name
```
Comment (piotrpdev) [inline pipelines/README.md:178]:
```suggestion
> **Note**
> If you are using a branch other than `main` for your gitops workflow, you will need to modify the `PipelineRuns` examples under `tekton/gitops-update-pipeline/example-pipelineruns/` to set the `gitRepoBranchBase` parameter to your custom branch name
```

Looks better:

> **Note**
> If you are using a branch other than `main` for your gitops workflow, you will need to modify the `PipelineRuns` examples under `tekton/gitops-update-pipeline/example-pipelineruns/` to set the `gitRepoBranchBase` parameter to your custom branch name

**Review by piotrpdev — APPROVED** [review]

File: pipelines/README.md (hunk 1 above)
Comment (grdryn) [reply pipelines/README.md:178]:
```suggestion
> [!NOTE]
> If you are using a branch other than `main` for your gitops workflow, you will need to modify the `PipelineRuns` examples under `tekton/gitops-update-pipeline/example-pipelineruns/` to set the `gitRepoBranchBase` parameter to your custom branch name
```

@piotrpdev Reading between the lines of the update from 21 July 2023 in the description of [this issue](https://github.com/orgs/community/discussions/16925), my understanding is that this is the more up-to-date syntax for this beta feature, and will replace the older syntax that you suggested. It says that "the old syntax will continue to work for some time", which I read as "we're not going to support the old syntax forever".

**Review by grdryn — APPROVED** [review]

**Comment by adelton** [comment]
Looks good.

It would be even nicer if the pipeline could default to the primary branch without having it specified manually but that would likely need to be a change in https://github.com/tektoncd/catalog/tree/main/task/github-open-pr/0.2 to do an extra API call to figure out the base branch if not specified.

**Comment by grdryn** [comment]
Merging this as it has 2 approvals. Suggestion can be optionally added asynchronously.

**Comment by adelton** [comment]
I've filed https://github.com/opendatahub-io/ai-edge/issues/141 to get the same done for the ACM part.

**Pull request merged**